presto-main/src/main/java/io/prestosql/sql/gen/JoinCompiler.java

```diff
@@ -679,7 +679,6 @@ private void generatePositionNotDistinctFromRowWithPageMethod(
             if (type.getJavaType().equals(Slice.class)) {
                 switch (type.getTypeSignature().getBase()) {
                     case StandardTypes.CHAR:
-                    case StandardTypes.IPADDRESS:
```

Comment:
While `IPADDRESS` is handled here,  the function `isDistinctFrom(Block left, ...)`  is a dead code, so why to change it ?  I would also add `UUID` and completely remove `isDistinctFrom(Block left, ...)` functions

Reply:
I guess it is a hack and we need to move to correspoding`Type` instead of handling here.

Reply:
The logic is similar for both `JoinCompiler` and `isDistinctFrom`